internal/status/ing_status.go

```diff
@@ -360,18 +363,25 @@ func deleteObject(option UpdateOptions, key string, isVSDelete bool, retryNum ..
 			if status.Hostname != host {
 				continue
 			}
-			if !lib.ValidateIngressForClass(key, mIngress) || !utils.CheckIfNamespaceAccepted(option.ServiceMetadata.Namespace) || !utils.HasElem(hostListIng, host) || isVSDelete {
+			if !lib.ValidateIngressForClass(key, mIngress) ||
+				!utils.CheckIfNamespaceAccepted(option.ServiceMetadata.Namespace) ||
+				!utils.HasElem(hostListIng, host) ||
+				isVSDelete ||
+				mIngress.GetDeletionTimestamp() != nil {
 				mIngress.Status.LoadBalancer.Ingress = append(mIngress.Status.LoadBalancer.Ingress[:i], mIngress.Status.LoadBalancer.Ingress[i+1:]...)
 			} else {
 				utils.AviLog.Debugf("key: %s, msg: skipping status deletion since host is present in the ingress: %v", key, host)
 			}
 		}
 	}
 
-	if len(mIngress.Status.LoadBalancer.Ingress) > 0 && mIngress.Status.LoadBalancer.Ingress[0].IP != "" {
-		lib.CheckAndSetIngressFinalizer(mIngress)
-	} else {
-		lib.RemoveIngressFinalizer(mIngress)
+	// Add/Remove Ingress finalizers only when operating in a VCF Cluster.
+	if utils.IsVCFCluster() {
```

Comment:
@chauhanshubham : I think we need to add `if vcfcluster` condition over here also: https://github.com/chauhanshubham/load-balancer-and-ingress-services-for-kubernetes/blob/finalizer-nonvcf/internal/nodes/route_ingress_model.go#L173